docs/proposals/volume-health-monitoring.md

```diff
@@ -0,0 +1,266 @@
+# Proposal: Supporting for CSI volume health feature 
+
+Last updated: 24.12.2021
+
+## Abstract
+
+This proposal contains approaches for supporting CSI volume health feature, using Kubernetes volume health monitor mechanism. 
+
+## Background
+
+Currently, Baremetal CSI does not support any general conventional way to monitor Persistent Volumes after they are provisioned in Kubernetes.
+This makes it very hard to debug and detect root causes when something happens to the volumes. 
+An application may detect that it can no longer write to volumes. However, the root cause happened at the underlying storage 
+system level. This requires several teams jointly debug and find out what has triggered the problem. 
+It could be that the volume runs out of capacity and needs an expansion. 
+It could be that the volume was deleted by accident outside of Kubernetes. 
```

Comment:
This case is partially covered by missing operational state.

Reply:
And again, it's about some logic over it, not standartized way to handle it with k8s native resources